packages/@react-spectrum/datepicker/docs/TimeField.mdx

```diff
@@ -0,0 +1,258 @@
+
+```tsx example
+import {getLocalTimeZone} from '@internationalized/date';
+import {useDateFormatter} from '@adobe/react-spectrum';
+
+function Example() {
+  let [date, setDate] = React.useState(parseAbsoluteToLocal('2021-04-07T18:45:22Z'));
+  let formatter = useDateFormatter({dateStyle: 'long', timeStyle: 'long'});
+
+  return (
+    <>
+      <TimeField label="Time" value={date} onChange={setDate} />
+      <p>Selected date and time: {formatter.format(date.toDate(getLocalTimeZone()))}</p>
+    </>
+  );
+}
+```
+
+## Validation
+
+`TimeField` can display a validation state to communicate to the user whether the current value is valid or invalid. Implement your own validation logic in your app and pass either `"valid"` or `"invalid"` via the `validationState` prop. The `errorMessage` prop can be used to communicate errors to the user.
+
+This example validates that the selected time is on a 15 minute increment.
+
+```tsx example
+function Example() {
+  let [time, setTime] = React.useState(new Time(9, 15));
+
+  return (
+    <TimeField
+      label="Meeting time"
+      value={time}
+      onChange={setTime}
+      validationState={time.minute % 15 ? 'invalid' : 'valid'}
+      description="Select a meeting time"
+      errorMessage="Meetings start every 15 minutes." />
+  );
+}
+```
+
+### Minimum and maximum values
+
+The `minValue` and `maxValue` props can also be used to perform builtin validation. This prevents the user from selecting dates outside the valid range in the calendar, and displays an invalid state if the user enters an invalid date into the date field.
+
+This example only accepts times between 9 AM and 5 PM.
+
+```tsx example
+<TimeField label="Meeting time" minValue={new Time(9)} maxValue={new Time(17)} />
+```
+
+## Props
+
+<PropTable component={docs.exports.TimeField} links={docs.links} />
+
+## Visual options
+
+### Quiet
+
+```tsx example
+<TimeField label="Event time" isQuiet />
+```
+
+### Disabled
+
+```tsx example
+<TimeField label="Event time" isDisabled />
+```
+
+### Read only
+
+The `isReadOnly` boolean prop makes the TimeField's value immutable. Unlike `isDisabled` the TimeField remains focusable.
+
+```tsx example
+<TimeField label="Event time" value={new Time(11)} isReadOnly />
+```
+
+### Label alignment and position
+[View guidelines](https://spectrum.adobe.com/page/text-field/#Label-position)
+
+By default, the label is positioned above the `TimeField`. The `labelPosition` prop can be used to position the label to the side. The `labelAlign` prop can be used to align the label as "start" or "end". For left-to-right (LTR) languages, "start" refers to the left most edge of the `TimeField` and "end" refers to the right most edge. For right-to-left (RTL) languages, this is flipped.
+
+```tsx example
+<TimeField label="Event time" labelPosition="side" labelAlign="end" />
```

Comment:
Why is this TimeField so much wider than the examples above or below?

Reply:
Might be a bug with side labels. Will look into it later